src/status_im/contexts/keycard/utils.cljs

```diff
@@ -16,33 +18,58 @@
    (re-matches #".*NFCError:100.*" error)))
 
 (defn validate-application-info
-  [profile-key-uid {:keys [key-uid paired? pin-retry-counter puk-retry-counter] :as application-info}]
-  (let [profile-mismatch? (or (nil? profile-key-uid) (not= profile-key-uid key-uid))]
-    (log/debug "[keycard]"              "login-with-keycard"
-               "empty application info" (empty? application-info)
-               "no key-uid"             (empty? key-uid)
-               "profile-mismatch?"      profile-mismatch?
-               "no pairing"             paired?)
-    (cond
-      (empty? application-info)
-      :keycard/error.not-keycard
-
-      (empty? (:key-uid application-info))
-      :keycard/error.keycard-blank
-
-      profile-mismatch?
-      :keycard/error.keycard-wrong
-
-      (not paired?)
-      :keycard/error.keycard-unpaired
-
-      (and (zero? pin-retry-counter)
-           (or (nil? puk-retry-counter)
-               (pos? puk-retry-counter)))
-      :keycard/error.keycard-frozen
-
-      (zero? puk-retry-counter)
-      :keycard/error.keycard-locked
-
-      :else
-      nil)))
+  [profile-key-uid
+   {:keys [key-uid has-master-key? paired? pin-retry-counter puk-retry-counter] :as application-info}]
+
+  (cond
+    (empty? application-info)
+    :keycard/error.not-keycard
+
+    (not has-master-key?)
+    :keycard/error.keycard-blank
+
+    (and (nil? profile-key-uid) (not= profile-key-uid key-uid))
+    :keycard/error.keycard-not-blank
+
+    (not= profile-key-uid key-uid)
+    :keycard/error.keycard-wrong-profile
```

Comment:
Please can you elaborate, how this check is different from above one?

Reply:
when we pass profile-key-uid nil, we expect empty card

Reply:
but yeah you're right it's not an error, `(and (nil? profile-key-uid) (not= profile-key-uid key-uid))` can be removed